### PR TITLE
Fixes #14188 - change pulp_parent to qpid_client class

### PR DESCRIPTION
**Pull request by johnpmitsch**

### CODE REVIEW

File: manifests/qpid_client.pp [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Pulp Master Certs configuration
+class certs::qpid_client (
+
+  $hostname   = $::certs::node_fqdn,
+  $generate   = $::certs::generate,
+  $regenerate = $::certs::regenerate,
+  $deploy     = $::certs::deploy,
+
+  $nodes_cert_dir   = $certs::params::nodes_cert_dir,
+  $nodes_cert_name  = $certs::params::nodes_cert_name,
```
Comment (ehelms) [inline manifests/qpid_client.pp]: Can drop these nodes

File: manifests/qpid_client.pp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    password_file => $certs::ca_key_password_file,
+  }
+
+  if $deploy {
+
+    file { $messaging_client_cert:
+      owner => 'apache',
+      group => 'apache',
+      mode  => '0640',
+    } -> Class['pulp::config']
```
Comment (ehelms) [inline manifests/qpid_client.pp]: This connection here shouldn't be needed anymore

Comment (ehelms) [inline manifests/qpid_client.pp]: You also need this part https://github.com/Katello/puppet-certs/pull/78/files#diff-17ce3af01e46365ea61a9b96d0f63574L69 as that is what sets up the cert

**Comment by stbenjam** [comment]
We need to pin rake to make tests pass, foreman did this here: https://github.com/theforeman/foreman-installer-modulesync/pull/28

I opened a PR for us to do it: https://github.com/Katello/foreman-installer-modulesync/pull/4

**Comment by stbenjam** [comment]
[test]

File: manifests/qpid_client.pp [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  $hostname   = $::certs::node_fqdn,
+  $generate   = $::certs::generate,
+  $regenerate = $::certs::regenerate,
+  $deploy     = $::certs::deploy,
+
+  $messaging_ca_cert     = $certs::ca_cert,
+  $messaging_client_cert = $certs::params::messaging_client_cert
+
+  ) inherits pulp::params { # lint:ignore:inherits_across_namespaces
```
Comment (ehelms) [inline manifests/qpid_client.pp]: We can drop this inherits

File: manifests/qpid_client.pp [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+  cert { "${::certs::qpid_client::hostname}-qpid-client-cert":
+    hostname      => $::certs::qpid_client::hostname,
+    common_name   => 'pulp-qpid-client-cert',
```
Comment (ehelms) [inline manifests/qpid_client.pp]: All instances of `$::certs::qpid_client::hostname` can be renamed to `$hostname` because its a local parameter

**Comment by ehelms** [comment]
I was thinking maybe we keep pulp_parent and pulp_child for now to make refactoring to use qpid_client less impactful and require less coordination of merging code.

File: manifests/qpid_client.pp [hunk 5]
```diff
@@ -0,0 +1,45 @@
+# Pulp Master Certs configuration
+class certs::qpid_client (
+
+  $hostname   = $::certs::node_fqdn,
+  $generate   = $::certs::generate,
+  $regenerate = $::certs::regenerate,
+  $deploy     = $::certs::deploy,
+
+  $messaging_ca_cert     = $certs::ca_cert,
```
Comment (ehelms) [inline manifests/qpid_client.pp]: Looks like this parameter isn't used and could be removed

**Comment by ehelms** [comment]
This will need updating as well https://github.com/Katello/puppet-certs/blob/master/manifests/capsule.pp

File: manifests/capsule.pp [hunk 6]
```diff
@@ -26,7 +26,6 @@
   class { '::certs::apache':        hostname => $capsule_fqdn }
   class { '::certs::qpid':          hostname => $capsule_fqdn }
   class { '::certs::qpid_router':   hostname => $capsule_fqdn }
-  class { '::certs::pulp_child':    hostname => $capsule_fqdn }
```
Comment (ehelms) [inline manifests/capsule.pp]: Need to replace this with qpid_client, and down below

**Comment by johnpmitsch** [comment]
@ehelms updated

**Comment by ehelms** [comment]
Can you fix the test failure? Otherwise looks good

**Comment by johnpmitsch** [comment]
@ehelms fixed

**Comment by johnpmitsch** [comment]
normal functionality and capsule-certs-generate is working with these changes

**Comment by ehelms** [comment]
ACK - thanks @johnpmitsch !

**Pull request merged**